cms-publish-rest/src/main/java/se/simonsoft/cms/publish/rest/PublishResource.java

```diff
@@ -223,7 +231,39 @@ public Response getReleaseJson(@QueryParam("item") CmsItemIdArg itemId,
 		return response;
 	}
 	
-	
+	/**
+	 * @param itemId
+	 * @param 
+	 * @return JSON containing started execution ID and presigned urls (same as webhook).
+	 * @throws Exception
+	 */
+	@POST
+	@Path("api/start")
+	@Consumes(MediaType.APPLICATION_JSON)
+	@Produces(MediaType.APPLICATION_JSON)
+	public Response doStartApi(@QueryParam("item") CmsItemIdArg itemId, String body) throws Exception {
+		
+		logger.debug("Start publication requested with item: {} and options: '{}'", itemId, body);
+		
+		// Deserialize the body.
+		ObjectMapper objectMapper = new ObjectMapper();
+		PublishStartOptions options = objectMapper.readValue(body, PublishStartOptions.class);
+
+		// Generate and log a UUID, set into options.executionid regardless if set already.
+		UUID uuid = UUID.randomUUID();
+		options.setExecutionid(uuid.toString());
+		logger.debug("Generated executionid: '{}'", uuid);
+
+		LinkedHashMap<String, String> result = publishStartService.doPublishStartItem(itemId, options);
+
+		GenericEntity<LinkedHashMap<String, String>> ge = new GenericEntity<LinkedHashMap<String, String>>(result) {};
```

Comment:
Probably don't need GenericEntity if we perform serialization here. The GenericEntity trick is needed when Wink resolves a REST writer implementation.

Reply:
Removed.